samples/drivers/adc/src/main.c

```diff
@@ -52,19 +36,46 @@ struct adc_channel_cfg channel_cfg = {
 	.differential = 0
 };
 
+struct adc_sequence_options options = {
+	.callback = adc_callback,
+};
+
 struct adc_sequence sequence = {
+	.options = &options,
 	/* individual channels will be added below */
-	.channels    = 0,
-	.buffer      = sample_buffer,
+	.channels = 0,
+	.buffer = sample_buffer,
 	/* buffer size in bytes, not number of samples */
 	.buffer_size = sizeof(sample_buffer),
-	.resolution  = ADC_RESOLUTION,
+	.resolution = ADC_RESOLUTION,
 };
 
+enum adc_action adc_callback(const struct device *dev,
+			     const struct adc_sequence *sequence,
+			     uint16_t sampling_index)
+{
+
+	for (uint8_t i = 0; i < ADC_NUM_CHANNELS; i++) {
+		uint16_t raw_value = ((uint16_t *)sequence->buffer)[i];
+
+		/*
+		 * Convert raw reading to millivolts if driver
+		 * supports reading of ADC reference voltage
+		 */
+		int32_t mv_value = raw_value;
+
+		adc_raw_to_millivolts(adc_ref_internal(dev), ADC_GAIN,
+				      ADC_RESOLUTION, &mv_value);
+		LOG_ERR("ADC ch%d reading %d mV", i, mv_value);
```

Comment:
We should move back to `printk` as the sample used originally.
```suggestion
		printk("ADC ch%d reading %d mV", i, mv_value);
```

Reply:
done